pkg/rbac/authenticationMiddleware.go

```diff
@@ -43,7 +44,20 @@ func AuthenticateUser(next http.Handler) http.Handler {
 		}
 
 		klog.V(6).Info("User authentication successful!")
+		//if we are authenticated can move on to next step: authorize: get cluster scopedresources and cache
+		AuthorizeUser(clientToken)
```

Comment:
`server.go` is a better place for calling `AuthorizeUser()`. This way we can keep each middleware function independent.

```go
	// Add authentication middleware to the /searchapi (ContextPath) subroute.
	apiSubrouter := router.PathPrefix(config.Cfg.ContextPath).Subrouter()
	apiSubrouter.Use(rbac.AuthenticateUser)
	// if we are authenticated can move on to next step: authorize: get cluster scopedresources and cache
	apiSubrouter.Use(rbac.AuthorizeUser)
	apiSubrouter.Handle("/graphql", srv.Handler)
```

Then, use the request context to share the `clienToken` across the different middleware functions.
Reference: https://stackoverflow.com/questions/39946583/how-to-pass-context-in-golang-request-to-middleware

```go
type ContextKey string
const ContextAuthTokenKey ContextKey = "authToken"


// Add clientToken to request context.
ctx := context.WithValue(r.Context(), ContextAuthTokenKey, clientToken)
// Continue executing the middleware chain.
next.ServeHTTP(w, r.WithContext(ctx))
```

Access the clientToken from AuthorizeUser()
```go
clientToken := r.Context().Value(ContextAuthTokenKey).(string)
```

Reply:
this is what I had initially but figured we are only getting the resources not authorizing yet.

Reply:
and was not not sure how to pass token from middlewares but the above works thank you!